src/audio/SDL_mixer.c

```diff
@@ -168,24 +165,22 @@ SDL_MixAudioFormat(Uint8 * dst, const Uint8 * src, SDL_AudioFormat format,
         {
             Sint16 src1, src2;
             int dst_sample;
-            const int max_audioval = ((1 << (16 - 1)) - 1);
-            const int min_audioval = -(1 << (16 - 1));
+            const int max_audioval = SDL_MAX_SINT16;
+            const int min_audioval = SDL_MIN_SINT16;
 
             len /= 2;
             while (len--) {
-                src1 = ((src[0]) << 8 | src[1]);
+                src1 = SDL_SwapBE16(*(Sint16 *)src);
                 ADJUST_VOLUME(src1, volume);
-                src2 = ((dst[0]) << 8 | dst[1]);
+                src2 = SDL_SwapBE16(*(Sint16 *)src);
                 src += 2;
                 dst_sample = src1 + src2;
                 if (dst_sample > max_audioval) {
                     dst_sample = max_audioval;
                 } else if (dst_sample < min_audioval) {
                     dst_sample = min_audioval;
                 }
-                dst[1] = dst_sample & 0xFF;
-                dst_sample >>= 8;
-                dst[0] = dst_sample & 0xFF;
+                *(Sint16 *)dst = SDL_SwapBE16(dst_sample);
```

Comment:
Possible alignment issues here (and at other similar changes below)?

Reply:
Is that a question to me? I think SDL_SwapBE16 should take care of the correct order, if that is the question.

Reply:
I meant unaligned writes to dst -- or am I missing something?

Reply:
it is a 16-bit audio, if it is not aligned for 16-bit read/write, then something is really off.

Reply:
> it is a 16-bit audio, if it is not aligned for 16-bit read/write, then something is really off.

Doesn't matter if it's 16bit audio, if the data has allocated as bytes you can't cast the U8 pointer to a U16 pointer, you will get alignment issues and invoke undefined behaviour.

Reply:
There is more problems, SDL_SwapLE16 and SDL_SwapBE16 will not swap anything if the native order is the same, which actually changes what the code does.

Reply:
@FredrikRosenqvist : Nope. You are wrong. Check case AUDIO_S32LSB.

Reply:
@pionere The alignment issue still exists though.

Reply:
@FredrikRosenqvist : what alignment issue is there which does not exist in case of AUDIO_S32LSB?